### ISSUE TITLE
Filtering even numbers in vector couses segmentation fault

### ISSUE THREAD

**Issue by lordofinsomnia** [ljubobratovicrelja/cpprange#1]
```c++
    const int vElements = 10;
    std::vector<int> v;
    for(auto i=0u;i<vElements;++i){
        v.push_back(i);
    }

    filter([] (auto e) { return e%2==0;}, v)
        .each([](auto e) {
            std::cout << e << std::endl;
        });
```

Vector of 10 numbers, and i want to printout just even numbers.
This is output:
```0
2
4
6
8
33
0
6409440
```

**Comment by ljubobratovicrelja** [comment]
Tnx, should be fixed with 8e996caf57209cb4c4e6b9f769761563e0df0063. Added example [here](https://github.com/ljubobratovicrelja/cpprange/blob/master/src/main.cpp#L136-L144).

**Comment by lordofinsomnia** [comment]
It works. :)
